Create indexes before loading

AustinSMueller opened the pull request:
Bento462 - creates indexes for each of the entries in the "id_fields" section of the properties files if the index does not already exist

AustinSMueller committed:
Create indexes before loading

Bento462 - creates indexes for each of the entries in the "id_fields" section of the properties files if the index does not already exist

n2iw reviewed (changes requested):
Two small changes needed:

1. We need to check both "tokenNames" and "properties" to match our  "node" and "id field". In case there is an index on a node, but on another property, we still need to create index for "id field" in that case.
2. I want to add a capability to specify indexes on other properties (instead of "id field") in property files. So that I can add additional indexes to nodes. I can provide more detail via Slack.

AustinSMueller committed:
Expanded Index Creation Functionality

Made changes according to feedback of code review.

Allowed for specification of non-id indexes and expanded checks to detect the possibility of an existing index with multiple properties

AustinSMueller commented:
Corrected issues found during review, the program now identifies existing indexes by checking the node name and all properties and also allows for the creation of specified non-id field indexes
